inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -358,5 +372,40 @@ public void putTrainedModel(Recommender aRecommender, Object aModel)
         {
             trainedModels.put(aRecommender.getId(), aModel);
         }
+
+        public void removePredictions(Recommender aRecommender, Project aProject)
+        {
+            // Remove incoming predictions
+            Predictions incoming = incomingPredictions.get(aProject.getId());
+            if (incoming != null) {
+                incoming.removePredictions(aRecommender.getId());
+                incomingPredictions.put(aProject.getId(), incoming);
+            }
+
+            // Remove active predictions
+            Predictions active = activePredictions.get(aProject.getId());
+            if (active != null) {
+                active.removePredictions(aRecommender.getId());
+                activePredictions.put(aProject.getId(), active);
+            }
+
+            // Remove trainedModel
+            trainedModels.remove(aRecommender.getId());
+
+            // Remove from activeRecommenders map
+            MultiValuedMap<AnnotationLayer, Recommender> newActiveRecommenders
+                = new HashSetValuedHashMap<>();
+            MapIterator<AnnotationLayer, Recommender> it
+                = activeRecommenders.mapIterator();
+
+            while (it.hasNext()) {
+                AnnotationLayer layer = it.next();
+                Recommender rec = it.getValue();
+                if (!rec.equals(aRecommender)) {
+                    newActiveRecommenders.put(layer, rec);
+                }
+            }
+            setActiveRecommenders(newActiveRecommenders);
```

Comment:
Normally, we only set the incoming model and let the rendering code turn the incoming model into the active model. If it is really valid to set the active model directly here, there should be a comment explaining why this is necessary.

Reply:
Never mind - wrong issue.